### ISSUE TITLE
File lock RuntimeError

### ISSUE THREAD

**Issue by hexrom** [tristanlatr/WPWatcher#66]
```
Traceback (most recent call last):
  File "/home/ec2-user/.local/bin/wpwatcher", line 8, in <module>
    sys.exit(main())
  File "/home/ec2-user/.local/lib/python3.9/site-packages/wpwatcher/cli.py", line 72, in main
    exit_code, reports = wpwatcher.run_scans()
  File "/home/ec2-user/.local/lib/python3.9/site-packages/wpwatcher/core.py", line 230, in run_scans
    self._run_scans(self.wp_sites)
  File "/home/ec2-user/.local/lib/python3.9/site-packages/wpwatcher/core.py", line 207, in _run_scans
    self.new_reports.append(f.result())
  File "/usr/lib64/python3.9/concurrent/futures/_base.py", line 446, in result
    return self.__get_result()
  File "/usr/lib64/python3.9/concurrent/futures/_base.py", line 391, in __get_result
    raise self._exception
  File "/usr/lib64/python3.9/concurrent/futures/thread.py", line 58, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/home/ec2-user/.local/lib/python3.9/site-packages/wpwatcher/core.py", line 179, in _scan_site
    self.wp_reports.write([wp_report])
  File "/home/ec2-user/.local/lib/python3.9/site-packages/wpwatcher/db.py", line 111, in write
    raise RuntimeError("The file lock must be acquired before writing data. ")
RuntimeError: The file lock must be acquired before writing data. 
```

I don't understand what is tripping it off or how to suppress this error?

**Comment by tristanlatr** [comment]
Hi @tacticthreat,

Thanks for the report, it looks like there is an issue with one of the latest version of [py-filelock](https://github.com/tox-dev/py-filelock/releases). Can you please try with older versions of `py-filelock` and report whether it's working ad with which version ? Thanks.

**Comment by tristanlatr** [comment]
It works with `filelock<=3.10.7`

**Comment by tristanlatr** [comment]
This issue is fixed in version 3.0.6
